reconcile/test/test_utils_oc.py

```diff
@@ -431,3 +431,100 @@ def test_clusters_errors_with_include_errors(
+        cluster_1 = {
+            'name': 'cl1',
+            'serverUrl': 'http://localhost',
+            'clusterAdminAutomationToken': {
+                'path': 'some-path',
+                'field': 'some-field'
+            },
+            'automationToken': {
+                'path': 'some-path',
+                'field': 'some-field'
+            }
+        }
+        cluster_2 = {
+            'name': 'cl2',
+            'serverUrl': 'http://localhost',
+            'clusterAdminAutomationToken': {
+                'path': 'some-path',
+                'field': 'some-field'
+            },
+            'automationToken': {
+                'path': 'some-path',
+                'field': 'some-field'
+            }
+        }
+        namespace_1 = {
+            'name': 'ns1',
+            'clusterAdmin': True,
+            'cluster': cluster_1
+
+        }
+
+        namespace_2 = {
+            'name': 'ns2',
+            'cluster': cluster_2
+
+        }
+
+        oc_map = OC_Map(namespaces=[namespace_1, namespace_2])
+
+        self.assertEqual(oc_map.clusters(), ["cl1", "cl2"])
+        self.assertEqual(oc_map.clusters(privileged=True), ["cl1"])
+
+        # both clusters are present as non privileged clusters in the map
+        self.assertIsInstance(oc_map.get(cluster_1['name']), OC)
+        self.assertIsInstance(oc_map.get(cluster_2['name']), OC)
+
+        # only cluster_1 is present as privileged cluster in the map
+        self.assertIsInstance(
+            oc_map.get(cluster_1['name'], privileged=True),
+            OC
+        )
+        self.assertIsInstance(
+            oc_map.get(cluster_2['name'], privileged=True),
+            OCLogMsg
+        )
+
+    @patch.object(reconcile.utils.oc, 'OC', autospec=True)
+    @patch.object(SecretReader, 'read', autospec=True)
+    def test_missing_cluster_automation_token(self, mock_secret_reader,
+                                              mock_oc):
+        cluster_1 = {
+            'name': 'cl1',
+            'serverUrl': 'http://localhost',
+            'automationToken': {
+                'path': 'some-path',
+                'field': 'some-field'
+            }
+        }
+        namespace_1 = {
+            'name': 'ns1',
+            'clusterAdmin': True,
+            'cluster': cluster_1
+
+        }
+
+        oc_map = OC_Map(namespaces=[namespace_1])
+
+        # check that non-priv OC got instantiated but priv one not
+        self.assertEqual(oc_map.clusters(), ["cl1"])
+        self.assertEqual(oc_map.clusters(privileged=True), [])
+        self.assertEqual(
+            oc_map.clusters(include_errors=True, privileged=True),
+            [cluster_1['name']]
+        )
+
+        self.assertIsInstance(
+            oc_map.get(cluster_1['name']),
+            OC
+        )
+        self.assertIsInstance(
+            oc_map.get(cluster_1['name'], privileged=True),
+            OCLogMsg
+        )
```

Comment:
Considering that the code uses this as a boolean, wouldn't it make sense to do `self.assertFalse(oc_map.get(cluster_1['name'], privileged=True))`
?

Reply:
fixed in 2e55eb1